### PR TITLE
hal: nuvoton: add NuMaker RTC controller driver

### PR DESCRIPTION
**Pull request by cyliangtw**
This PR is to add RTC controller into CMakeList for M46x and M2l31x series.

### CODE REVIEW

File: m2l31x/Devices/M2L31/Include/system_M2L31.h [hunk 1]
```diff
@@ -27,7 +27,7 @@ extern "C" {
 /*----------------------------------------------------------------------------
   Define clocks
  *----------------------------------------------------------------------------*/
-#define __HXT       (32000000UL)    //(12000000UL)    /*!< External Crystal Clock Frequency     */
+#define __HXT       (12000000UL)    /*!< External Crystal Clock Frequency     */
```
Comment (fabiobaltieri) [inline m2l31x/Devices/M2L31/Include/system_M2L31.h:30]: did you mean to just replace the number rather than pushing the old one on the side

Comment (cyliangtw) [reply m2l31x/Devices/M2L31/Include/system_M2L31.h:30]: Yes, 32M is the temporary crystal setting and 12M is the permanent setting & sync with proprietary BSP.

Comment (cyliangtw) [reply m2l31x/Devices/M2L31/Include/system_M2L31.h:30]: if you have any concern and prefer to push the old one on the side, I could modify as:
#define __HXT       (12000000UL)    //(32000000UL)

Comment (fabiobaltieri) [reply m2l31x/Devices/M2L31/Include/system_M2L31.h:30]: Sorry I've misread the diff and then forgot to followup, did not realize you were *removing* the temporary setting -- this is fine thanks for clarifying.

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**